src/lib/spinel/radio_spinel.cpp

```diff
@@ -1585,7 +1585,11 @@ void RadioSpinel::HandleTransmitDone(uint32_t          aCommand,
     static_cast<Mac::TxFrame *>(mTransmitFrame)->SetIsHeaderUpdated(headerUpdated);
 
     if ((sRadioCaps & OT_RADIO_CAPS_TRANSMIT_SEC) && headerUpdated &&
-        static_cast<Mac::TxFrame *>(mTransmitFrame)->GetSecurityEnabled())
+        static_cast<Mac::TxFrame *>(mTransmitFrame)->GetSecurityEnabled()
+#if OPENTHREAD_CONFIG_DIAG_ENABLE
+        && (!mDiagMode)
```

Comment:
I think we should check `otRadioFrame.mInfo.mTxInfo.mIsSecurityProcessed`?

Reply:
Ignore the previous comment. It looks like this block should be skipped if the `mIsHeaderUpdated` was `true`, i.e. not changed from `false` to `true` by the coprocessor.

Reply:
Updated